src/main/java/com/beowulfe/hap/impl/services/AbstractServiceImpl.java

```diff
@@ -39,13 +27,14 @@ public AbstractServiceImpl(String type) {
 	 *
 	 * @param type unique UUID of the service. This information can be obtained from HomeKit Accessory Simulator.
 	 * @param accessory HomeKit accessory exposed as a service.
+	 * @param serviceName name of the service. This information is usually the name of the accessory.
      */
-	public AbstractServiceImpl(String type, HomekitAccessory accessory) {
```

Comment:
I think we should leave the old constructor. Not necessarily for API compatibility, but because I think most use cases would probably just use the accessory name. The old constructor can just call the new one with accessory.getLabel()